File: algorithms/java/303_Range_Sum_Query_Immutable.java
class NumArray {
    int[] dynmNums;
    
    public NumArray(int[] nums) {
        dynmNums = new int[nums.length + 1];
        dynmNums[0] = 0;
        for(int i = 1 ; i <= nums.length ; ++i) {
            dynmNums[i] = dynmNums[i-1] + nums[i-1];
        }
    }
    
    public int sumRange(int i, int j) {
        return dynmNums[j+1] - dynmNums[i];
    }
}

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * int param_1 = obj.sumRange(i,j);
 */

File: algorithms/python/303_Range_Sum_Query_Immutable.py
class NumArray:

    def __init__(self, nums: List[int]):
        self.dynmNums = [0];
        for num in nums:
            self.dynmNums.append(self.dynmNums[-1] + num);
        self.dynmNums = self.dynmNums[1:] + [0];
        
    def sumRange(self, i: int, j: int) -> int:
        return self.dynmNums[j] - self.dynmNums[i-1]
        


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(i,j)
